core/storage/suggestion/gae_models.py

```diff
@@ -152,93 +157,23 @@ def create(
             score_category=score_category).put()
 
     @classmethod
-    def get_suggestions_by_type(cls, suggestion_type):
-        """Gets all suggestions of a particular type.
-
-        Args:
-            suggestion_type: str. The type of the suggestions.
-
-        Returns:
-            list(SuggestionModel). A list of suggestions of the given
-                type, up to a maximum of feconf.DEFAULT_QUERY_LIMIT
-                suggestions.
-        """
-        return cls.get_all().filter(
-            cls.suggestion_type == suggestion_type).fetch(
-                feconf.DEFAULT_QUERY_LIMIT)
-
-    @classmethod
-    def get_suggestions_by_author(cls, author_id):
-        """Gets all suggestions created by the given author.
+    def query_suggestions(cls, queries):
+        """Queries for suggestions.
 
         Args:
-            author_id: str. The ID of the author of the suggestion.
+            queries: list(tuple). A list of queries. The first element in each
+                tuple is the field to be queried, and the second element is its
+                value.
 
         Returns:
-            list(SuggestionModel). A list of suggestions by the given author,
-            up to a maximum of feconf.DEFAULT_QUERY_LIMIT suggestions.
+            list(SuggestionModel). A list of suggestions that match the given
+            query values, up to a maximum of feconf.DEFAULT_QUERY_LIMIT
+            suggestions.
         """
-        return cls.get_all().filter(
-            cls.author_id == author_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
-
-    @classmethod
-    def get_suggestions_assigned_to_reviewer(cls, assigned_reviewer_id):
-        """Gets all suggestions assigned to the given user for review.
+        query = cls.query()
```

Comment:
You shouldn't be calling the function arg "queries" and this QueryObject "query". This suggests that the former is just a plural version of the latter, which is not true.

Perhaps the arg should be called query_fields_and_values? But you want to propagate this name change up the stack as well.

Reply:
Done.